kv/task.go

```diff
@@ -315,13 +315,15 @@ func (s *Service) findTasksByUser(ctx context.Context, tx Tx, filter influxdb.Ta
 			continue
 		}
 
-		if taskFilterMatch(filter.Type, task.Type) {
+		matchFn := newTaskMatchFn(filter, nil)
```

Comment:
This should be moved outside the `for` loop and the second argument can be passed the `org` variable, which is an instance of `*influxdb.Organization`. The `matchFn` will then ensure the org is filtered if set. 

Doing that will also allow you to remove lines 314-316:

```go
if org != nil && task.OrganizationID != org.ID {
	continue
}
```